xskillscore/core/probabilistic.py

```diff
@@ -488,32 +549,53 @@ def rps(
     """Calculate Ranked Probability Score.
 
      .. math::
-        RPS(p, k) = 1/M \\sum_{m=1}^{M}
-        [(\\sum_{k=1}^{m} p_k) - (\\sum_{k=1}^{m} o_k)]^{2}
+        RPS = \\sum_{m=1}^{M}[(\\sum_{k=1}^{m} y_k) - (\\sum_{k=1}^{m} o_k)]^{2}
+
+    where ``y`` and ``o`` are forecast and observation probabilities in ``M``
+    categories.
+
+     .. note::
+        Takes the sum over all categories as in Weigel et al. 2007 and not the mean as
+        in https://www.cawcr.gov.au/projects/verification/verif_web_page.html#RPS.
+        Therefore RPS has no upper boundary.
 
     Parameters
     ----------
     observations : xarray.Dataset or xarray.DataArray
-        The observations or set of observations of the event.
-        Data should be boolean or logical \
-        (True or 1 for event occurance, False or 0 for non-occurance).
+        The observations of the event.
+        Further requirements are specified based on ``category_edges``.
     forecasts : xarray.Dataset or xarray.DataArray
-        The forecast likelihoods of the event.
-        If ``fair==False``, forecasts should be between 0 and 1 without a dimension
-        ``member_dim`` or should be boolean (True,False) or binary (0, 1) containing a
-        member dimension (probabilities will be internally calculated by
-        ``.mean(member_dim))``. If ``fair==True``, forecasts must be boolean
-        (True,False) or binary (0, 1) containing dimension ``member_dim``.
-    category_edges : array_like
-        Category bin edges used to compute the CDFs. Similar to np.histogram, \
-        all but the last (righthand-most) bin include the left edge and exclude \
-        the right edge. The last bin includes both edges.
+        The forecast of the event with dimension specified by ``member_dim``.
+        Further requirements are specified based on ``category_edges``.
+    category_edges : array_like, xr.Dataset, xr.DataArray, None
+        Category edges used to compute the CDFs. Similar to np.histogram, all but the
+        last (righthand-most) bin include the left edge and exclude the right edge. The
+        last bin includes both edges.
+        These ``category_edge`` must span the full range of the observations and
+        forecasts distribution. Forecasts, observations and category_edge are expected
+        in absolute units or probabilities consistently.
+        ``category_edges`` decides how xs.rps interprets forecasts and observations.
+
+        - np.array (1d): will be internally converted and broadcasted to observations.
+
+        - xr.Dataset/xr.DataArray: edges of the categories provided
+          as dimension ``category_edge`` with optional category labels as
+          ``category_edge`` coordinate. Use xr.Dataset/xr.DataArray if edges
+          multi-dimensional and vary across dimensions.
```

Comment:
"Use this if your category edges vary across dimensions of forecasts and observations, but are the same for both."